### PR TITLE
feat: @stacks/ui take 2

### PR DESCRIPTION
**Pull request by aulneau**
> [Download the latest build](https://github.com/blockstack/stacks-wallet/actions/runs/774046133)

### What this does
- now uses `@stacks/ui` in place of the older `@blockstack/ui`
- refactored the modal to use the newer `ControlledModal` component
- dark mode!
Screen Shot 2021-04-20 at 2 30 22 PM

### CODE REVIEW

File: app/components/back-button.tsx [hunk 1]
```diff
@@ -36,13 +31,15 @@ export const BackButton: FC<Omit<ButtonProps, 'children'>> = ({ onClick, ...prop
         handleBack();
         onClick?.(e);
       }}
+      pointerEvents={!hasBackState ? 'none' : 'all'}
       isDisabled={!hasBackState}
```
Comment (kyranjamie) [inline app/components/back-button.tsx:35]: image

this causes this runtime error

**Review by kyranjamie — COMMENTED** [review]
What's the cause of all the `React does not recognise the 'xxx' prop on a DOM element`? 

image

File: app/components/back-button.tsx [hunk 2]
```diff
@@ -18,14 +18,9 @@ export const BackButton: FC<Omit<ButtonProps, 'children'>> = ({ onClick, ...prop
...
   return (
-    <Button
-      variant="unstyled"
-      _hover={{ background: 'none' }}
-      position="absolute"
-      left={0}
-      top="5px"
-      height="34px"
-      width="34px"
+    <IconButton
```
Comment (kyranjamie) [inline app/components/back-button.tsx:21]: `<IconButton />` isn't a button 🤔  This change results in no focus state for the back button.

File: app/components/back-button.tsx (hunk 1 above)
Comment (aulneau) [reply app/components/back-button.tsx:35]: ah yeah should be removed, or we can do `as={Button}`

File: app/components/back-button.tsx (hunk 2 above)
Comment (aulneau) [reply app/components/back-button.tsx:21]: as="button" or as={Button}

**Comment by aulneau** [comment]
> What's the cause of all the `React does not recognise the 'xxx' prop on a DOM element`?
> 
> image

With the update of the UI lib, both types and props are a bit improved, which means if previously we had let some probs down to the base dom element, the old version didn't warn about it, or checked if they weren't true dom attrs and removed them for us.

**Comment by markmhx** [comment]
The dark mode toggle doesn't appear for me using the MacOS testnet build linked in the description:

image

**Comment by aulneau** [comment]
> The dark mode toggle doesn't appear for me using the MacOS testnet build linked in the description:

Can you try again?

**Comment by kyranjamie** [comment]
As soon as all runtime errors are fixed, let's merge into `fix/update-electron`. 

From there I will fix merges with my refactor PR https://github.com/blockstack/stacks-wallet/pull/751 (please review), and that should be the base for QA.

**Comment by markmhx** [comment]
So I should wait to QA dark mode here until that branch is merged?

**Pull request closed without merging**